modules/network_synchronizer/data_buffer.cpp

```diff
@@ -477,6 +489,13 @@ Vector2 DataBuffer::read_normalized_vector2(CompressionLevel p_compression_level
 Vector3 DataBuffer::add_vector3(Vector3 p_input, CompressionLevel p_compression_level) {
 	ERR_FAIL_COND_V(is_reading == true, p_input);
 
+#ifndef REAL_T_IS_DOUBLE
+	// Fallback to compression level 1 if real_t is float
+	if (p_compression_level == DataBuffer::COMPRESSION_LEVEL_0) {
+		p_compression_level = DataBuffer::COMPRESSION_LEVEL_1;
+	}
+#endif
```

Comment:
Why don't put this fallback directly inside the float compressor function? I think it would make more sense anyway.

Also, I think we need to add a warning here, so the user knows that doesn't make sense use compression 0 when it's using a single precision binary.

Reply:
Because `add_real` does not depend on this definition. It accepts doubles. But Vector2 and Vector3 use `real_t` as underlying type.

You will have an incredible amount of warnings because this function gets called very often.

Reply:
Yes but it doesn't make senses anyway compress a single precision to double to convert it back to single precision.

The warning would tell the user how to properly use the api, and it would spam only when the api is not properly used. Which make sense. What do you think?

Reply:
> Yes but it doesn't make senses anyway compress a single precision to double to convert it back to single precision.

It depends on what type is used. If double is used, and not `real_t`, then there will be no narrowing conversion.
GDScript documentation says that [64-bit floats are used](https://docs.godotengine.org/ru/stable/getting_started/scripting/gdscript/gdscript_basics.html#float) and only some types have (such as Vector2) uses `real_t`.

> The warning would tell the user how to properly use the api, and it would spam only when the api is not properly used. Which make sense. What do you think?

Hm... Okay, I will add a warning!

Reply:
Added.

Reply:
Typo here too.